### ISSUE TITLE
Unexpected failover when cluster down and a slave is start first

### ISSUE THREAD

**Issue by jaceklubzinski** [signal18/replication-manager#306]
Hi,

during testing one of the database cluster failures scenarios I ran into an unexpected condition.

Tested on
- mrm latest official docker image: Replication-Manager v2.1.1-139-g68c63d18
- MariaDB latest official 10.2 docker image

With two database servers (db1 and db2) configured in master slave topology I stopped db1 and db2 and next start db2.
Expected behavior: mrm should set db2 as slave and wait for master
Actual behavior: mrm electing db2 as new master

The same situation occurred with `failover-restart-unsafe = false` in config file. 

Date | Log level | Log info
-- | -- | --
2020/09/09 09:25:51 | INFO | Master switch on 172.16.2.22 complete
2020/09/09 09:25:51 | INFO | Switching other slaves to the new master
2020/09/09 09:25:51 | INFO | Inject fake transaction on new master 172.16.2.22
2020/09/09 09:25:49 | INFO | Waiting 2s for unmanaged proxy to monitor route change
2020/09/09 09:25:49 | INFO | Failover proxies
2020/09/09 09:25:49 | INFO | Resetting slave on new master and set read/write mode on
2020/09/09 09:25:49 | INFO | Stopping slave threads on new master
2020/09/09 09:25:49 | DEBUG | Candidate was in sync=false
2020/09/09 09:25:49 | DEBUG | master_log_pos=446
2020/09/09 09:25:49 | DEBUG | master_log_file=mysqld-bin.000001
2020/09/09 09:25:49 | DEBUG | Save replication status before electing
2020/09/09 09:25:49 | INFO | Reading all relay logs on 172.16.2.22
2020/09/09 09:25:49 | INFO | Waiting for candidate master to apply relay log
2020/09/09 09:25:49 | INFO | Slave 172.16.2.22 has been elected as a new master
2020/09/09 09:25:49 | INFO | Election matrice: [ 	{ 		"URL": "172.16.2.22", 		"Indice": 0, 		"Pos": 1000000000446, 		"Seq": 1, 		"Prefered": false, 		"Ignoredconf": false, 		"Ignoredrelay": false, 		"Ignoredmultimaster": false, 		"Ignoredreplication": false, 		"Weight": 0 	} ]
2020/09/09 09:25:48 | INFO | Electing a new master
2020/09/09 09:25:48 | INFO | ------------------------
2020/09/09 09:25:48 | INFO | Starting master failover
2020/09/09 09:25:48 | INFO | ------------------------
2020/09/09 09:25:48 | INFO | Enforce read only on slave 172.16.2.22
2020/09/09 09:25:48 | DEBUG | Remove cookie remove /var/lib/replication-manager/ppms-cluster-main/172.16.2.21_3306/@cookie_waitstop: no such file or directory
2020/09/09 09:25:45 | ERROR | No master found from replication

Config file (tested also with default config file):
```
[cluster-main]
title = "cluster-main"
db-servers-hosts = "172.16.2.21,172.16.2.22"
db-servers-credential = "root:root"
replication-credential = "root:root"
db-servers-prefered-master = "172.16.2.21"
[DEFAULT]
monitoring-datadir = "/var/lib/replication-manager"
monitoring-sharedir = "/usr/share/replication-manager"
log-file = "/dev/stdout"
log-level = 2
force-slave-gtid-mode = true
force-slave-readonly = true
http-bind-address = "0.0.0.0"
http-port = 10001
http-root = "/usr/share/replication-manager/dashboard"
http-server = true
http-session-lifetime = 3600
failover-mode = "automatic"
monitoring-ticker = 2
check-type = "tcp"
failover-falsepositive-ping-counter = 5
failover-falsepositive-heartbeat = true
failover-falsepositive-heartbeat-timeout = 3
failcount-reset-time = 300
failover-readonly = true
failover-limit = 2
failover-time-limit = 60
failover-restart-unsafe = true
failover-max-slave-delay = 300
switchover-max-slave-delay = 30
mariadb-binary-path = "/usr/share/replication-manager"
```

**Comment by svaroqui** [comment]
Hi thanks, 

This function is returning no server 

func (cluster *Cluster) GetMasterFromReplication(s *ServerMonitor) (*ServerMonitor, error) {

	for _, server := range cluster.Servers {
		if server.ServerID == s.ServerID {
			//Ignoring same ServerID
			continue
		}
		if len(s.Replications) > 0 {

			if cluster.Conf.LogLevel > 2 {
				cluster.LogPrintf(LvlDbg, "GetMasterFromReplication server  %d  lookup if server %s is the one : %d", s.GetReplicationServerID(), server.URL, server.ServerID)
			}
			if s.IsIOThreadRunning() && s.IsSQLThreadRunning() {
				if s.GetReplicationServerID() == server.ServerID {
					return server, nil
				}
			} else {
				if s.GetReplicationMasterHost() == server.Host && s.GetReplicationMasterPort() == server.Port {
					return server, nil
				}
			}
		}

	}

	return nil, nil
}

Can you send me a show slave status to eliminate  that  s.GetReplicationMasterHost() == server.Host && s.GetReplicationMasterPort() == server.Port would be different aka IP vs host name ?

**Comment by jaceklubzinski** [comment]
Show slave status from db-2 (172.16.2.22)
```
root@16b297414748:/# mysql -uroot -proot

MariaDB [(none)]> show slave status\G;
*************************** 1. row ***************************
               Slave_IO_State: Waiting for master to send event
                  Master_Host: 172.16.2.21
                  Master_User: root
                  Master_Port: 3306
                Connect_Retry: 5
              Master_Log_File: mysqld-bin.000001
          Read_Master_Log_Pos: 446
               Relay_Log_File: mysqld-relay-bin.000002
                Relay_Log_Pos: 672
        Relay_Master_Log_File: mysqld-bin.000001
             Slave_IO_Running: Yes
            Slave_SQL_Running: Yes
              Replicate_Do_DB: 
          Replicate_Ignore_DB: 
           Replicate_Do_Table: 
       Replicate_Ignore_Table: 
      Replicate_Wild_Do_Table: 
  Replicate_Wild_Ignore_Table: 
                   Last_Errno: 0
                   Last_Error: 
                 Skip_Counter: 0
          Exec_Master_Log_Pos: 446
              Relay_Log_Space: 982
              Until_Condition: None
               Until_Log_File: 
                Until_Log_Pos: 0
           Master_SSL_Allowed: No
           Master_SSL_CA_File: 
           Master_SSL_CA_Path: 
              Master_SSL_Cert: 
            Master_SSL_Cipher: 
               Master_SSL_Key: 
        Seconds_Behind_Master: 0
Master_SSL_Verify_Server_Cert: No
                Last_IO_Errno: 0
                Last_IO_Error: 
               Last_SQL_Errno: 0
               Last_SQL_Error: 
  Replicate_Ignore_Server_Ids: 
             Master_Server_Id: 1
               Master_SSL_Crl: 
           Master_SSL_Crlpath: 
                   Using_Gtid: Slave_Pos
                  Gtid_IO_Pos: 0-1-1
      Replicate_Do_Domain_Ids: 
  Replicate_Ignore_Domain_Ids: 
                Parallel_Mode: conservative
                    SQL_Delay: 0
          SQL_Remaining_Delay: NULL
      Slave_SQL_Running_State: Slave has read all relay log; waiting for the slave I/O thread to update it
```

**Comment by svaroqui** [comment]
i was able to reproduce that issue

**Comment by jaceklubzinski** [comment]
is there any workaround for this issue? Additional database/mrm configuration?

**Comment by svaroqui** [comment]
Hi,

Fixed in commit b1370d31 please can you test with CI build before i push to repositories

Tx
/svar

**Comment by jaceklubzinski** [comment]
Hi,
work as expected now. Thanks for quick fix!
